File: Deploy/RapierLoom.Examples/08-Monitoring/VB/Library.vb
Public Class Library

    Private Function calcFibonacci(ByVal a As UInt64) As UInt64
        If (a > 40) Then
            Throw New ApplicationException("Error: Can only compute the first 40 fibonacci numbers.")
        End If
        If (a <= 2) Then
            Return CULng(1)
        End If
        Return (Me.calcFibonacci((a - 1)) + Me.calcFibonacci((a - 2)))
    End Function

    Public Overridable Function Fibonacci(ByVal a As UInt64) As UInt64
        Return Me.calcFibonacci(a)
    End Function

End Class

File: Deploy/GripperLoom.Examples/08-Monitoring/CS/Library.cs
// ----------------------------------------------------------------
// * The LOOM.NET project http://rapier-loom.net *
// (C) Copyright by Wolfgang Schult. All rights reserved. 
// This code is licensed under the Apache License 2.0
// To get more information, go to http://loom.codeplex.com/license
// ----------------------------------------------------------------
using System;
using System.Collections.Generic;
using System.Text;

namespace Monitoring
{
    [MonitoringAspect]
    public class Library
    {
        public virtual ulong Fibonacci(ulong a)
        {
            return calcFibonacci(a);
        }


        #region private methods - not matched

        private ulong calcFibonacci(ulong a)
        {
            if (a > 40)
                throw new ApplicationException("Error: Can only compute the first 40 fibonacci numbers.");

            if (a <= 2)
                return 1;
            else
                return (calcFibonacci(a - 1) + calcFibonacci(a - 2));
        }

        #endregion

    }
}
